src/deadline_worker_agent/startup/bootstrap.py

```diff
@@ -520,6 +566,28 @@ def _enforce_no_instance_profile_or_stop_worker(
         raise
 
 
+def _get_instance_id() -> str | None:
+    """
+    This function attempts to query the IMDS /instance-id endpoint for the instance ID.
+    The query will return the instance ID on success, and will return None if the query fails or
+    times out. The timeout is limited to 5 seconds and will occur if running off AWS.
```

Comment:
nit: documentation wasn't updated after last rev:

```suggestion
    times out. The timeout is limited to half a second and will occur if running off AWS.
```